internal/compiler/program.go

```diff
@@ -152,6 +153,8 @@ func NewProgram(options ProgramOptions) *Program {
 	p.files, p.resolvedModules = processAllProgramFiles(p.host, p.programOptions, p.compilerOptions, p.resolver, rootFiles, libs)
 	p.filesByPath = make(map[tspath.Path]*ast.SourceFile, len(p.files))
 	for _, file := range p.files {
+		packageType := p.resolver.GetPackageJsonTypeIfApplicable(string(file.Path()))
```

Comment:
In the old compiler, we avoid looking up the package.json if the file has a `.mts`/`.cts`/`.mjs`/`.cjs` extension. And I wonder if this can be done concurrently in the file loader? There’s already a mutex protecting some map writes there.